include/xtensor/xcontainer.hpp

```diff
@@ -220,7 +190,7 @@ namespace xt
         using inner_strides_type = typename base_type::inner_strides_type;
         using inner_backstrides_type = typename base_type::inner_backstrides_type;
 
-        void reshape(const shape_type& shape);
+        void reshape(const shape_type& shape, bool force = false);
         void reshape(const shape_type& shape, layout l);
         void reshape(const shape_type& shape, const strides_type& strides);
```

Comment:
I find it strange to provide methods that may fail to compile, depending the template parameters you use. Most of the times, people need the `reshape` overload that takes the new shape only, other overloads are provided for tuning / optimization. Enabling these last ones for dynamic layout only doesn't make the interface more inconsistent thant providing reshape methods for containers only, for instance.

Maybe the cleanest way would be to provide these methods in a specialization of the `strided_container` class, but that would be a bit of overkill imo.